Class/Practise 2.1/tests.py

```diff
@@ -0,0 +1,100 @@
+import subprocess
+import sys
+import os
+
+def run_program(program_path, *args):
+    result = subprocess.run([program_path, *args], capture_output=True, text=True)
+    return result.stdout.strip(), result.stderr.strip()
+
+def test_case_1(program_path):
+    # Step 1: Ensure fresh start
+    if os.path.exists("user_stats.txt"):
+        os.remove("user_stats.txt")
+    
+    # Step 2: Run with "Alice"
+    output, error = run_program(program_path, "Alice")
+    assert output == "Welcome, Alice!"
+    
+    # Step 3: Run again with "Alice"
+    output, error = run_program(program_path, "Alice")
+    assert output == "Hello again(x2), Alice!"
+    print("Test Case 1 Passed")
+
+def test_case_2(program_path):
+    # Step 1: Run with "Bob"
+    run_program(program_path, "Bob")
+    
+    # Step 2: Run again with "Bob"
+    run_program(program_path, "Bob")
```

Comment:
The reason why I did not assert the output here because the second test is aimed at checking "delete" functionality as is shown in next lines:
`    # Step 3: Reset statistics
    output, error = run_program(program_path, "Bob", "delete")
    assert output == "Statistics for Bob have been reset."
    
    # Step 4: Run again with "Bob"
    output, error = run_program(program_path, "Bob")
    assert output == "Welcome, Bob!"
    print("Test Case 2 Passed")
`

On the other hand the first test is checking Welcome and counting so there is no need to duplicate assertions here